core/sqf/src/seatrans/tm/hbasetmlib2/javaobjectinterfacetm.cpp

```diff
@@ -478,3 +514,23 @@ void JavaObjectInterfaceTM::appendExceptionMessages(JNIEnv *jenv, jthrowable a_e
     jenv->DeleteLocalRef(a_exception);
 }
 
+short JavaObjectInterfaceTM::getExceptionErrorCode(JNIEnv *jenv, jthrowable a_exception)
+{
+  jshort errCode = JOI_OK;
+  if(a_exception != NULL)
+  {
+    //Check to see if this exception has getErrorCode method.
+    //Only custom subclasses have errorcode defined.
```

Comment:
Consider caching the methodId or it is an expensive way to get the error code. Is this method called only when error is present